server.js

```diff
@@ -785,41 +797,97 @@ cache(function (data, match, sendBadge, request) {
   });
 }));
 
+/**
+ * Registers a service in the shields.io system.
+ *
+ * `config` is an object containing these keys:
+ *  - name: Name of the service
+ *  - type: Type of the service (build, downloads, version, etc)
+ *  - uri: Regular expression to use for URIs for this service's badges
+ *  - examples: Example URIs for this service. These should use the format
+ *    specified in `uri`, and can be used to demonstrate how to use badges for
+ *    this service.
+ *  - handler: Asynchronous function to handle requests for this service. This
+ *    function takes two arguments:
+ *    1. Capture groups from the `uri` regular expression
+ *    2. A `request` method for sending cached HTTP requests to third-party
+ *       servers.
+ */
+function registerService(config) {
+  // Regular expressions treat "/" specially, so we need to escape them
+  const escapedPath = config.uri.replace(/\//g, '\\/');
+  const fullRegex = '^' + escapedPath + '.(svg|png|gif|jpg|json)$';
+
+  camp.route(new RegExp(fullRegex),
+  cache(function(data, match, sendBadge, request) {
+    // Assumes the final capture group is the extension
+    const format = match.pop();
+    // Discard the first capture group as it's the entire match
+    const params = match.slice(1);
+
+    const badgeData = getBadgeData(config.type, data);
+    config.handler(params, request.asPromise)
+      .then(serviceData => {
```

Comment:
This is still using promise syntax since I didn't want to touch the implementation of `cache` for now.